core/src/main/scala/org/apache/spark/memory/StaticMemoryManager.scala

```diff
@@ -53,6 +53,20 @@ private[spark] class StaticMemoryManager(
     (maxStorageMemory * conf.getDouble("spark.storage.unrollFraction", 0.2)).toLong
   }
 
+  override def acquireStorageMemory(
+      blockId: BlockId,
+      numBytes: Long,
+      evictedBlocks: mutable.Buffer[(BlockId, BlockStatus)]): Boolean = synchronized {
+    if (numBytes > storageMemoryPool.poolSize) {
```

Comment:
if you make the other one (in UMM) `numBytes > maxStorageMemory` then it would make sense to do the same here, so it's more consistent (small readability improvement)